packages/firestore/test/unit/local/indexeddb_schema.test.ts

```diff
@@ -145,10 +147,65 @@ describe('IndexedDbSchema: createOrUpgradeDb', () => {
   });
 
   it('can upgrade from schema version 2 to 3', () => {
-    return withDb(2, async () => {}).then(() =>
-      withDb(3, async db => {
+    const testWrite = { delete: 'foo' };
+    const testMutations = [
+      {
+        userId: 'foo',
+        batchId: 0,
+        localWriteTime: 1337,
+        mutations: []
+      },
+      {
+        userId: 'foo',
+        batchId: 1,
+        localWriteTime: 1337,
+        mutations: [testWrite]
+      },
+      {
+        userId: 'foo',
+        batchId: 42,
+        localWriteTime: 1337,
+        mutations: [testWrite, testWrite]
+      }
+    ];
+
+    return withDb(2, db => {
+      const sdb = new SimpleDb(db);
+      return sdb.runTransaction('readwrite', [DbMutationBatch.store], txn => {
+        const store = txn.store(DbMutationBatch.store);
+        let p = PersistencePromise.resolve();
+        for (const testMutation of testMutations) {
+          p = p.next(() => store.put(testMutation));
+        }
+        return p;
```

Comment:
Could you do something like this?

```
return PersistencePromise.waitFor(testMutations.map(mutation => store.put(mutation)));
```

Don't care too much though...

Reply:
Yup. Done.